os/net/lwip/src/include/lwip/sockets.h

```diff
@@ -157,6 +157,8 @@ struct lwip_setgetsockopt_data {
 	err_t err;
 	/** semaphore to wake up the calling task */
 	void *completed_sem;
+	/** pid of current task */
+	pid_t pid;
```

Comment:
why did you add this?

Reply:
when lwip_setsockopt() is called then it also calls lwip_setsockopt_impl() and for lwip_setsockopt_impl() pid was changed. so i am storing the pid in data in lwip_setsockopt().